File: testng-core/src/test/java/test/hook/samples/ConfigurableFailureWithStatusAlteredSample.java
package test.hook.samples;

import static test.hook.HookableTest.HOOK_INVOKED_ATTRIBUTE;

import org.testng.IConfigureCallBack;
import org.testng.ITestResult;
import org.testng.annotations.Test;

public class ConfigurableFailureWithStatusAlteredSample extends BaseConfigurableSample {

  @Override
  public void run(IConfigureCallBack callBack, ITestResult testResult) {
    testResult.setAttribute(HOOK_INVOKED_ATTRIBUTE, "true");
    // Not calling the callback
    testResult.setStatus(ITestResult.SUCCESS);
  }

  @Test
  public void hookWasNotRun() {}
}

File: testng-core/src/test/kotlin/test/hook/samples/ConfigurableFailureWithStatusAlteredSample.kt
package test.hook.samples

import test.hook.HOOK_INVOKED_ATTRIBUTE

import org.testng.IConfigureCallBack
import org.testng.ITestResult
import org.testng.annotations.Test

class ConfigurableFailureWithStatusAlteredSample : BaseConfigurableSample() {

    override fun run(callBack: IConfigureCallBack, testResult: ITestResult) {
        testResult.setAttribute(HOOK_INVOKED_ATTRIBUTE, "true")
        // Not calling the callback
        testResult.status = ITestResult.SUCCESS
    }

    @Test
    fun hookWasNotRun() {
    }
}
